x/axelarnet/message_handler.go

```diff
@@ -21,24 +22,46 @@ import (
 
 // Fee is used to pay relayer for executing cross chain message
 type Fee struct {
-	Amount    string `json:"amount"`
-	Recipient string `json:"recipient"`
+	Amount          string  `json:"amount"`
+	Recipient       string  `json:"recipient"`
+	RefundRecipient *string `json:"refund_recipient"`
 }
 
-func (f Fee) validate(ctx sdk.Context, n types.Nexus, b types.BankKeeper, token sdk.Coin, msgType nexus.MessageType) error {
-	amt, ok := sdk.NewIntFromString(f.Amount)
-	if !ok || amt.LTE(sdk.ZeroInt()) {
+// ValidateBasic validates the fee
+func (f Fee) ValidateBasic() error {
+	amount, ok := sdk.NewIntFromString(f.Amount)
+	if !ok || !amount.IsPositive() {
 		return fmt.Errorf("invalid fee amount")
 	}
 
-	afterFee := token.Amount.Sub(amt)
+	if _, err := sdk.AccAddressFromBech32(f.Recipient); err != nil {
+		return err
+	}
+
+	if f.RefundRecipient != nil {
+		// The refund recipient is an address of the chain where the request is originally initiated from,
+		// and therefore we cannot validate much of it.
+		if err := utils.ValidateString(*f.RefundRecipient); err != nil {
```

Comment:
not running the chain specific address validator over it?

Reply:
That's a stateful check, which can't be done in `ValidateBasic` but down there in `validateFee` at https://github.com/axelarnetwork/axelar-core/blob/8b43dd3e48e2800efacb80ac22eb0ccb2744ed41/x/axelarnet/message_handler.go#L81